app/gui/src/model/suggestion_database/entry.rs

```diff
@@ -232,9 +248,32 @@ impl Entry {
         is_method && not_extension
     }
 
-    /// Get the full qualified name of the entry.
-    pub fn qualified_name(&self) -> tp::QualifiedName {
-        tp::QualifiedName::new_module_member(self.module.clone(), self.name.clone())
+    /// Get the segments of the full qualified name of the entry.
+    pub fn qualified_name_segments(&self) -> QualifiedNameSegments {
+        fn collect_segments<'a, I>(iter: I) -> QualifiedNameSegments
+        where I: Iterator<Item = &'a str> {
+            QualifiedNameSegments(iter.map(NameSegment::new).collect())
+        }
+        fn collect_segments_and_entry_name<'a, I>(
```

Comment:
If you want to do something like that, implement `FromIterator` for `QualifiedNameSegments` instead.